game_for_class.py

```diff
@@ -71,15 +99,84 @@ def attack(self,
         :param players: TODO: Fill in
         :return: TODO: Fill in
         """
+
+
+            
+
+class Ally(Character):
+    def __init__(self,
+                 name: str,
+                 hp: float,
+                 attack: float,
+                 defense: float,
+                 speed: float):
+        super().__init__(
+            name=name,
+            hp=hp,
+            attack=attack,
+            defense=defense,
+            speed=speed,
+            team=1
+        )
+
+
+    def act(self, players: List ['Character']) -> List['Character']:
+        """Overides the default method
+
+        Allows a user to select a supported action (attack, block)
+        and also determine if a target is attacking
+
+        :param players
+        """
         all_enemy_locations = self.get_all_enemy_indices(players)
         if all_enemy_locations:
-            targeted_index = all_enemy_locations[0]
-            targeted_player = players[targeted_index]
-            damaged_player = self.deal_damage(targeted_player)
-            players[targeted_index] = damaged_player
-            return players
-        else:
-            return players
+            print('What will you do?')
+            print('"w" to attack or "s" to block')
+            move_options = {'w': self.attack(), 's': self.block()}
+            player_move = None
+            while player_move == None:
+                move = input('>')
+                player_move = move_options.get(move)
+        return player_move
+        
+
+class Enemy(Character):
+    def __init__(self,
+                 name: str,
+                 hp: float,
+                 attack: float,
+                 defense: float,
+                 speed: float):
+        super().__init__(
+            name=name,
+            hp=hp,
+            attack=attack,
+            defense=defense,
+            speed=speed,
+            team=2
+        )
+
+
+    def act(self, players):
+        """Overides the default act method
+
+        randomly selects an action (attack, block) and if attacking,
+        randomly selects a valid target.
+
+        :param players
+        """
+
+        
+        
+        all_enemy_locations = self.get_all_enemy_indices(players)
+        if all_enemy_locations:
+            move_options = ['attack', 'block']
+            move = random.choice(move_options)
+            if move == 'attack':
+                self.attack(other_player)
+            if move == 'block':
+                self.block(other_player)
+                return move
 
     def block(self, other_player: 'Character'):
```

Comment:
The method definition differs from `Character`'s implementation. You want them to take in the same arguments